### ISSUE TITLE
Setting Up Content-Security-Policy

### ISSUE THREAD

**Issue by Olusiji** [electron-userland/electron-webpack#159]
I get this warning in the console every time i compile my electron-app,
"Electron Security Warning (Insecure Content-Security-Policy) This renderer process has either no Content Security Policy set or a policy with "unsafe-eval" enabled. This exposes users of this app to unnecessary security risks."

I've tried to set-up Content Security by inserting the meta-tag into the header. But webpack generates the static index.html at run-time. So i'm not sure where or how to add content security policy.

Any help will be appreciated

**Comment by ramazanalkann** [comment]
Hi I am tried this <script src="bla bla"></script> but not worked.

And I tried this method This is working for me.
```
    <script>
      require('./ramo.js') // My js file
      require('./renderer.js')
   </script>
```

**Comment by markmorris** [comment]
Having this same issue, not sure how to fix, but my app does not sure when compiled because of this.

**Comment by fabiospampinato** [comment]
> I've tried to set-up Content Security by inserting the meta-tag into the header. But webpack generates the static index.html at run-time. So i'm not sure where or how to add content security policy.

I've added a way of overriding the default HTML template in #261, this should greatly help in solving this issue.

**Comment by dospolov** [comment]
`process.env['ELECTRON_DISABLE_SECURITY_WARNINGS'] = 'true'`
worked for me

**Comment by hexag0d** [comment]
@dospolov  I don't think that addresses the underlying security concern it just disables the warning

**Comment by hexag0d** [comment]
I used the advice of @ramazan-alkan  (thanks btw)

```
<!-- <script src="bla bla"></script> not worked. => -->
    <script>
      require('./ramo.js') // My js file
      require('./renderer.js')
   </script>
```

and I also had to do this so that the inline CSS styles would stop throwing the error; I searched online and couldn't find any information about why _css_ styles would be considered insecure.  I totally understand why inline JS would be insecure, but not inline css.  Anywhoo, this is the other part of what I had to do in order to fix the warning:
```
<head>
<meta http-equiv="Content-Security-Policy" content="default-src *; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline'">
 </head>
```

put that in the head tag and do what ramazan said

**Comment by kiruh** [comment]
# Solution

You can override the `electronWebpack.renderer.template` value in your `package.json`:

```json
"electronWebpack": {
  "renderer": {
    "template": "src/renderer/index.html"
  }
}
```

After that create `src/renderer/index.html` with desired `Content-Security-Policy`:

```html
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Hello World!</title>
    <meta http-equiv="Content-Security-Policy" content="script-src 'self' 'unsafe-inline';" />
</head>
<body style="background: white;">
    <h1>Hello World!</h1>
    <p>
        We are using Node.js <span id="node-version"></span>,
        Chromium <span id="chrome-version"></span>,
        and Electron <span id="electron-version"></span>.
    </p>
</body>
</html>
```

If you run `npm run dev` now, you'll get the following error:

> Uncaught EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self' 'unsafe-inline'".

To solve it, override the `electronWebpack.renderer.webpackConfig` value in your `package.json`:

```json
"electronWebpack": {
  "renderer": {
    "template": "src/renderer/index.html",
    "webpackConfig": "renderer.webpack.js"
  }
}
```

After that create `renderer.webpack.js` and change `devtool` option:

```js
module.exports = {
    devtool: "nosources-source-map",
};
```
Now if you run `npm run dev`, there will be neither warnings nor errors.

# Explanation

By default `electron-webpack` sets `devtool` option to `"eval-source-map"` in development, but `eval` operator is considered harmful according to [Electron's official documentation](https://www.electronjs.org/docs/tutorial/security#6-define-a-content-security-policy).

You can set `devtool` to one of [these options](https://webpack.js.org/configuration/devtool/) omitting those that use `eval` operator.
